packages/block-editor/src/hooks/align.js

```diff
@@ -138,7 +138,18 @@ function BlockEditAlignmentToolbarControls( {
 				nextAlign = '';
 			}
 		}
-		setAttributes( { align: nextAlign } );
+		/**
+		 * This filter exists so that the image block can reset
+		 * its aspect ratio and custom size when the alignment is set
+		 * to either wide or full.
+		 */
+		const filteredAttributes = applyFilters(
+			'block-library.image.alignmentUpdate',
+			blockName,
+			{ ...attributes, align: nextAlign }
```

Comment:
Looking at the way the `UPDATE_BLOCK_ATTRIBUTES` reducer works, I think it might be best to keep the object here just to the attributes that are being modified rather than all attributes.

Passing all causes an unnecessary performance hit, as the code loops through the attributes one by one.

Could still provide all attributes as a separate param as it was before, so that the filter function can make decisions based on the values.